frontend/packages/dev-console/src/components/catalog/hooks/useCatalogCategories.ts

```diff
@@ -1,6 +1,30 @@
+import * as React from 'react';
+
 import { defaultCatalogCategories } from '../utils/default-categories';
 import { CatalogCategories } from '../utils/types';
+import { mapCategories } from '../utils/custom-catalog-categories';
+
+const useCatalogCategories = (): CatalogCategories => {
+  const categories = React.useMemo<CatalogCategories>(() => {
+    try {
+      const categoriesString = window.SERVER_FLAGS.developerCatalogCategories;
+      if (!categoriesString) {
+        return defaultCatalogCategories;
+      }
+
+      const categoriesArray = JSON.parse(categoriesString);
+
+      if (!categoriesArray || !categoriesArray.length) {
```

Comment:
Seems valid to me if someone supplies an empty array or a null value that we hide all categories.

Reply:
Removed the second condition, catalog already skipped "All" and "Others" in this case. Result looks like this:

![image](https://user-images.githubusercontent.com/139310/100730989-16460f80-33cb-11eb-8b1c-4866a5717581.png)